pkg/provider/azure_utils.go

```diff
@@ -342,3 +341,152 @@ func extractVmssVMName(name string) (string, string, error) {
+// getIPFamiliesEnabled checks if IPv4, IPv6 are enabled according to svc.Spec.IPFamilies.
+func getIPFamiliesEnabled(svc *v1.Service) (v4Enabled bool, v6Enabled bool) {
+	for _, ipFamily := range svc.Spec.IPFamilies {
+		if ipFamily == v1.IPv4Protocol {
+			v4Enabled = true
+		} else {
+			v6Enabled = true
+		}
+	}
+	return
+}
+
+// getServiceLoadBalancerIP retrieves LB IP from IPv4 annotation, then IPv6 annotation, then service.Spec.LoadBalancerIP.
+func getServiceLoadBalancerIP(service *v1.Service, isIPv6 bool) string {
+	if service == nil {
+		return ""
+	}
+
+	if ip, ok := service.Annotations[consts.ServiceAnnotationLoadBalancerIPDualStack[isIPv6]]; ok && ip != "" {
+		return ip
+	}
+
+	// Retrieve LB IP from service.Spec.LoadBalancerIP (will be deprecated)
+	svcLBIP := service.Spec.LoadBalancerIP
+	if (net.ParseIP(svcLBIP).To4() != nil && !isIPv6) ||
+		(net.ParseIP(svcLBIP).To4() == nil && isIPv6) {
+		return svcLBIP
+	}
+	return ""
+}
+
+func getServiceLoadBalancerIPs(service *v1.Service) []string {
+	if service == nil {
+		return []string{}
+	}
+
+	ips := []string{}
+	if ip, ok := service.Annotations[consts.ServiceAnnotationLoadBalancerIPDualStack[false]]; ok && ip != "" {
+		ips = append(ips, ip)
+	}
+	if ip, ok := service.Annotations[consts.ServiceAnnotationLoadBalancerIPDualStack[true]]; ok && ip != "" {
+		ips = append(ips, ip)
+	}
+	if len(ips) != 0 {
+		return ips
+	}
+
+	lbIP := service.Spec.LoadBalancerIP
+	if lbIP != "" {
+		ips = append(ips, lbIP)
+	}
+
+	return ips
+}
+
+// setServiceLoadBalancerIP sets LB IP to a Service
+func setServiceLoadBalancerIP(service *v1.Service, ip string) {
+	if service.Annotations == nil {
+		service.Annotations = map[string]string{}
+	}
+	isIPv6 := net.ParseIP(ip).To4() == nil
+	service.Annotations[consts.ServiceAnnotationLoadBalancerIPDualStack[isIPv6]] = ip
+}
+
+func getServicePIPName(service *v1.Service, isIPv6 bool) string {
+	if service == nil {
+		return ""
+	}
+
+	if consts.DualstackSupported {
+		if name, ok := service.Annotations[consts.ServiceAnnotationPIPNameDualStack[isIPv6]]; ok && name != "" {
+			return name
+		}
+	}
+	return service.Annotations[consts.ServiceAnnotationPIPName]
+}
+
+func getServicePIPPrefixID(service *v1.Service, isIPv6 bool) string {
+	if service == nil {
+		return ""
+	}
+
+	if consts.DualstackSupported {
+		if name, ok := service.Annotations[consts.ServiceAnnotationPIPPrefixIDDualStack[isIPv6]]; ok && name != "" {
+			return name
+		}
+	}
+	return service.Annotations[consts.ServiceAnnotationPIPPrefixID]
+}
+
+func getResourceByIPFamily(resource string, isIPv6 bool) string {
+	if !consts.DualstackSupported {
+		return resource
+	}
+
+	if isIPv6 {
+		return fmt.Sprintf("%s-%s", resource, v6Suffix)
+	}
+	return fmt.Sprintf("%s-%s", resource, v4Suffix)
+}
+
+// ifFIPIPv6 checks if the frontend IP configuration is of IPv6.
+func (az *Cloud) ifFIPIPv6(fip *network.FrontendIPConfiguration, pips *[]network.PublicIPAddress, isInternal bool) (isIPv6 bool, err error) {
```

Comment:
s/ifFIPIPv6/isFIPIPv6

and replace `if` with `is` in other function names.

Reply:
Updated.